generate/generate.go

```diff
@@ -904,11 +904,14 @@ func (g Generator) AddBindMount(bind string) error {
 }
 
 // SetupPrivileged sets up the priviledge-related fields inside g.spec.
-func (g Generator) SetupPrivileged(privileged bool) {
+func (g Generator) SetupPrivileged(privileged bool, hostSpecific bool) {
 	if privileged {
 		// Add all capabilities in privileged mode.
 		var finalCapList []string
 		for _, cap := range capability.List() {
+			if hostSpecific && cap > capability.CAP_LAST_CAP {
+				continue
+			}
 			finalCapList = append(finalCapList, fmt.Sprintf("CAP_%s", strings.ToUpper(cap.String())))
```

Comment:
On Thu, Jul 21, 2016 at 03:03:04PM -0700, hmeng-19 wrote:
> I would rather keep this function as it is for code efficiency.

Are we really worried about performance for compiling a capability
list?  Can you spell out your concern in more detail or give a dummy
benchmark or something?